sympy/functions/elementary/trigonometric.py

```diff
@@ -1006,7 +1006,7 @@ def _eval_is_zero(self):
         if pi_mult:
             return fuzzy_and([(pi_mult - S.Half).is_integer, rest.is_zero])
         else:
-            return rest.is_zero
+            return (rest - S.Half).is_zero
```

Comment:
I don't think that this is correct e.g.:
```python
In [1]: e = cos(Rational(1, 2))

In [2]: e
Out[2]: cos(1/2)

In [3]: e.is_zero
Out[3]: True

In [4]: e.evalf()
Out[4]: 0.877582561890373
```

Reply:
Thanks, I'll fix it right away